File: packages/database/src/catalog.ts
export type CatalogStockState = 'in_stock' | 'low_stock' | 'out_of_stock' | 'preorder';

export interface CatalogCategory {
  id: string;
  slug: string;
  name: string;
  description: string;
  heroImage: string;
  eyebrow?: string;
}

export interface CatalogFeaturedStory {
  title: string;
  subtitle: string;
  thumbnail: string;
  href: string;
}

export interface CatalogProductColor {
  name: string;
  hex: string;
  gallery: string[];
}

export interface CatalogProductSummary {
  id: string;
  categorySlug: string;
  name: string;
  slug: string;
  price: number;
  comparePrice?: number | null;
  image: string;
  swatchCount: number;
  sizes: string[];
  badge?: string;
  stock: number;
  stockState?: CatalogStockState;
  isFeatured?: boolean;
  tags?: string[];
}

export interface CatalogProductDetail extends CatalogProductSummary {
  colors: CatalogProductColor[];
  description: string[];
  materials: string[];
  care: string[];
  trustBadges?: string[];
}

export interface CatalogLink {
  label: string;
  href: string;
}

export interface StorefrontFooterData {
  company: string;
  address: string;
  phone: string;
  shoppingHelp: CatalogLink[];
  about: CatalogLink[];
  policy: CatalogLink[];
  social: CatalogLink[];
}

export interface StorefrontQuickLink {
  eyebrow: string;
  title: string;
  description: string;
  href: string;
}

export interface StorefrontAccountHighlight {
  label: string;
  value: string;
  description: string;
}

export interface StorefrontCartItem {
  id: string;
  categorySlug: string;
  productSlug: string;
  name: string;
  price: number;
  image: string;
  quantity: number;
  color: string;
  size: string;
}

export interface StorefrontCatalogPayload {
  categories: CatalogCategory[];
  featuredStories: CatalogFeaturedStory[];
  products: CatalogProductDetail[];
  searchPrompts: string[];
  utilityQuickLinks: StorefrontQuickLink[];
  accountHighlights: StorefrontAccountHighlight[];
  checkoutSteps: string[];
  trustSignals: string[];
  footer: StorefrontFooterData;
}

File: services/api/app/schemas/catalog.py
from pydantic import BaseModel, ConfigDict, Field


class CatalogSchema(BaseModel):
    model_config = ConfigDict(populate_by_name=True)


class CatalogCategory(CatalogSchema):
    id: str
    slug: str
    name: str
    description: str
    hero_image: str = Field(serialization_alias="heroImage")
    eyebrow: str | None = None


class CatalogFeaturedStory(CatalogSchema):
    title: str
    subtitle: str
    thumbnail: str
    href: str


class CatalogProductColor(CatalogSchema):
    name: str
    hex: str
    gallery: list[str]


class CatalogProductSummary(CatalogSchema):
    id: str
    category_slug: str = Field(serialization_alias="categorySlug")
    name: str
    slug: str
    price: int
    compare_price: int | None = Field(default=None, serialization_alias="comparePrice")
    image: str
    swatch_count: int = Field(serialization_alias="swatchCount")
    sizes: list[str]
    badge: str | None = None
    stock: int
    stock_state: str | None = Field(default=None, serialization_alias="stockState")
    is_featured: bool | None = Field(default=None, serialization_alias="isFeatured")
    tags: list[str] | None = None


class CatalogProductDetail(CatalogProductSummary):
    colors: list[CatalogProductColor]
    description: list[str]
    materials: list[str]
    care: list[str]
    trust_badges: list[str] | None = Field(
        default=None, serialization_alias="trustBadges"
    )
    compare_price: int | None = Field(default=None, serialization_alias="comparePrice")


class CatalogLink(CatalogSchema):
    label: str
    href: str


class StorefrontFooterData(CatalogSchema):
    company: str
    address: str
    phone: str
    shopping_help: list[CatalogLink] = Field(serialization_alias="shoppingHelp")
    about: list[CatalogLink]
    policy: list[CatalogLink]
    social: list[CatalogLink]


class StorefrontQuickLink(CatalogSchema):
    eyebrow: str
    title: str
    description: str
    href: str


class StorefrontAccountHighlight(CatalogSchema):
    label: str
    value: str
    description: str


class StorefrontCatalogPayload(CatalogSchema):
    categories: list[CatalogCategory]
    featured_stories: list[CatalogFeaturedStory] = Field(
        serialization_alias="featuredStories"
    )
    products: list[CatalogProductDetail]
    search_prompts: list[str] = Field(serialization_alias="searchPrompts")
    utility_quick_links: list[StorefrontQuickLink] = Field(
        serialization_alias="utilityQuickLinks"
    )
    account_highlights: list[StorefrontAccountHighlight] = Field(
        serialization_alias="accountHighlights"
    )
    checkout_steps: list[str] = Field(serialization_alias="checkoutSteps")
    trust_signals: list[str] = Field(serialization_alias="trustSignals")
    footer: StorefrontFooterData
